lib/pyfrc/test_support/pytest_plugin.py

```diff
@@ -59,6 +59,9 @@ def pytest_runtest_setup(self):
                        "If your robot class has an __init__ function, it must call super().__init__()!"
     
     def pytest_runtest_teardown(self, nextitem):
+        # Let any child threads run in realtime to allow cancelling if it
+        # has been implemented.
+        self._fake_time.freerun_children()
```

Comment:
This is an interesting issue - the necessity to let the threads run free is because TimerTasks have a call made to their cancel() method by the free() method in PIDController.
https://github.com/robotpy/robotpy-wpilib/blob/f489e60a373351904c13efba71e47e1606b74198/wpilib/wpilib/pidcontroller.py#L138-L144

If the thread isn't running at this point, this call will block on cancel()'s call to join(). But it won't necessarily be stopped on the next run through of the loop, so throwing an exception when it calls delay() would likely cause it to throw everytime.

For a thread that doesn't have a terminate mechanism, there is no method to call after the test terminates to stop it - so the reference to the object just gets dropped and the next test begins. If the thread doesn't have a way to stop and join it, I don't think there is much we can do.
What do you think?